src/include/txn.i

```diff
@@ -262,36 +259,32 @@ __wt_txn_update_set_timestamp(WT_SESSION_IMPL *session,
 	wt_timestamp_t *timestamp;
 	bool needs_timestamp;
 
-	txn = &session->txn;
-	upd = op->u.op_upd;
 	if (was_set != NULL)
 		*was_set = false;
 
-	/*
-	 * The timestamp is in the page deleted structure for truncates, or
-	 * in the update for other operations.
-	 */
-	if (op->type == WT_TXN_OP_REF_DELETE)
-		timestamp = op->u.ref == NULL || op->u.ref->page_del == NULL ?
-		    NULL : &op->u.ref->page_del->timestamp;
-	else
-		timestamp = op->u.op_upd == NULL ?
-		    NULL : &op->u.op_upd->timestamp;
+	txn = &session->txn;
 
 	/*
 	 * Updates in the metadata never get timestamps (either now or at
 	 * commit): metadata cannot be read at a point in time, only the most
 	 * recently committed data matches files on disk.
 	 */
-	needs_timestamp = !WT_IS_METADATA(op->btree->dhandle) &&
-	    F_ISSET(txn, WT_TXN_HAS_TS_COMMIT) &&
-	    (timestamp == NULL || __wt_timestamp_iszero(timestamp) ||
-	    F_ISSET(txn, WT_TXN_PREPARE));
+	if (WT_IS_METADATA(op->btree->dhandle) ||
+	    !F_ISSET(txn, WT_TXN_HAS_TS_COMMIT))
```

Comment:
I don't believe it's OK to exit early if the transaction is in WT_TXN_PREPARE.

Reply:
@bvpvamsikrishna pointed out that a transaction in WT_TXN_PREPARE will always have a commit timestamp - so it's OK not to test it explicitly.

Reply:
> @bvpvamsikrishna pointed out that a transaction in WT_TXN_PREPARE will always have a commit timestamp

I think that's worth a comment.